.editorconfig

```diff
@@ -0,0 +1,33 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+# Matches multiple files with brace expansion notation
+# Set default charset
+[*.{js,css,html}]
+charset = utf-8
+
+# 2 space indentation
+[*.{js,json,yml,yaml,proto,sh,html,css}]
```

Comment:
I disagree. Library sh and bash scripts should have the `.sh` extension, but executable scripts should not. This is a subject style issue, but I'm old school on this, I guess (even though I haven't always been 100% consistent myself). The only convenience for adding the extension has to do with using editors that require the extension for syntax highlighting. The [Google Shell Style Guide](https://google.github.io/styleguide/shell.xml#File_Extensions) agrees. Bottom line, if the file has a shebang line and the executable bit set, then don't add an extension. If the script is meant to be invoked by another script and is not executable standalone, then add the extension.